### ISSUE TITLE
Passing a string as a page number raises IndexError instead of TypeError.

### ISSUE THREAD

**Issue by bwagner** [pymupdf/PyMuPDF#2961]
### Description of the bug

```
import fitz
doc = fitz.open("some.pdf")
doc["0"]
```
raises `IndexError`:
```
IndexError: page 0 not in document
```
Which is misleading, since the page 0 happens to be in the document, it's just being addressed wrongly.

### How to reproduce the bug

```
import fitz
doc = fitz.open("some.pdf")
doc["1"]
```
raises `IndexError`:
```
IndexError: page 0 not in document
```
Which is misleading, since the page 0 happens to be in the document, it's just being addressed wrongly.

### PyMuPDF version

1.23.9rc1

### Operating system

MacOS

### Python version

3.12

**Comment by bwagner** [comment]
I tried to [fix it](https://github.com/pymupdf/PyMuPDF/compare/main...bwagner:PyMuPDF:main), but my changes don't come through when I install the fixed package. When patching the installed file directly, however, my changes are effective. If anyone could recommend what I need to do to have my changes come through when installing, please let me know.

**Comment by julian-smith-artifex-com** [comment]
PyMuPDF in git changed a week or two ago to default to installing the rebased implementation as module `fitz`, not the classic implementation.

So you should probably be patching `src/__init__.py`, not `src_classic/fitz_old.i`.

[Apart from that, i'm wondering whether it might be better to simply do `assert isinstance(i, int)`, because this feels more appropriate for what is a programming error, not a normal runtime error?]

**Comment by bwagner** [comment]
Thank you!

**Comment by bwagner** [comment]
I signed the CLA. Is anything else required for this PR to be applied?

**Comment by julian-smith-artifex-com** [comment]
[Reopening until the fix is released.]

**Comment by julian-smith-artifex-com** [comment]
This is fixed in release candidate 1.23.9rc2; see: #2996

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.9.
